src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -60,12 +52,15 @@ internal class VisualStudioActiveDocumentTracker : ForegroundThreadAffinitizedOb
             : base(threadingContext, assertIsForeground: false)
         {
             _editorAdaptersFactoryService = editorAdaptersFactoryService;
-            _asyncTasks = new JoinableTaskCollection(threadingContext.JoinableTaskContext);
-            _asyncTasks.Add(ThreadingContext.JoinableTaskFactory.RunAsync(async () =>
+            ThreadingContext.RunWithShutdownBlockAsync(async cancellationToken =>
             {
-                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+                await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
                 var monitorSelectionService = (IVsMonitorSelection)await asyncServiceProvider.GetServiceAsync(typeof(SVsShellMonitorSelection)).ConfigureAwait(true);
+                Assumes.Present(monitorSelectionService);
```

Comment:
If a tree falls in the forest and no one is there to hear it, why did they bother to plant it? 😉

In this case, you're throwing an exception that no one will ever know about, since you drop the `JoinableTask` on the floor that is returned from `RunWithShutdownBlockAsync`. We generally recommend you to at least add `.FileAndForget` to the end of the method call so that you get telemetry fault events to find out when things fail.

Reply:
➡️ This is common right now. I filed #29273 previously to investigate solutions.